api/envoy/config/route/v3/route_components.proto

```diff
@@ -627,6 +631,43 @@ message CorsPolicy {
   core.v3.RuntimeFractionalPercent shadow_enabled = 10;
 }
 
+// The router is capable of shadowing traffic from one cluster to another. The current
+// implementation is "fire and forget," meaning Envoy will not wait for the shadow cluster to
+// respond before returning the response from the primary cluster. All normal statistics are
+// collected for the shadow cluster making this feature useful for testing.
+//
+// During shadowing, the host/authority header is altered such that *-shadow* is appended. This is
+// useful for logging. For example, *cluster1* becomes *cluster1-shadow*.
+//
+// .. note::
+//
+//   Shadowing will not be triggered if the primary cluster does not exist.
+message RequestMirrorPolicy {
```

Comment:
This is technically change the type, since it's now top-level rather than nested in `RouteAction`. We've been weary of this in https://github.com/envoyproxy/envoy/blob/main/api/API_VERSIONING.md, since it causes chaos in some places in general, e.g. when we have type URLs for extensions change. In this case, I think what is being done is safe and should not break any wire proto / JSON / YAML. It would potentially break some consuming code when type signature change.

My instinct is that it's relatively benign and the advantages (avoiding redundant definitions) outweigh any concerns. But would like to verify with other @envoyproxy/api-shepherds (in particular @markdroth and @mattklein123).

Reply:
@markdroth @mattklein123 can we get some feedback on the API please? I'll look at the rest of the impl, but would be nice to ensure we are on the same page API wise first. Thanks!

Reply:
Could we not use the one nested in RouteAction (with forward declaration or something similar) to avoid the controversy?

Reply:
Yes I would just leave it where it was and just reference that type. I think that's not that big of a deal and would avoid any potential compile issues for people.

Reply:
I move RequestMirrorPolicy back under RouteAction to keep api unchanged. Thanks for the feedback, htuch, rgs1 and mattklein123